os/compression/compress_read.c

```diff
@@ -379,14 +560,56 @@ int compress_init(int filfd, uint16_t offset, off_t *filelen)
 	/* Assign file length as that of uncompressed file */
 	*filelen = compression_header.binary_size;
 
+	/* Set number of blocks to use for caching */
+	if (CONFIG_DECOMPRESSION_CACHE_BLOCKS < (CUTOFF_RATIO_CACHE_BLOCKS) * (compression_header.sections - 1))
+		number_blocks_caching = CONFIG_DECOMPRESSION_CACHE_BLOCKS;
+	else
+		number_blocks_caching = ((CUTOFF_RATIO_CACHE_BLOCKS) * (compression_header.sections - 1));
+
+	/* Initialize max_accesed_count to 0 */
+	max_accessed_count = 0;
+
 #if CONFIG_COMPRESSION_TYPE == 1
-	/* Allocating memory for read and out buffer to be used for LZMA decompression */
+	/* Allocating memory for compressed_data_buffer and blockcache list to be used for LZMA decompression */
 	if (compression_header.compression_format == COMPRESSION_TYPE_LZMA) {
-		buffers.read_buffer = (unsigned char *)malloc(compression_header.blocksize + 5);
-		buffers.out_buffer = (unsigned char *)malloc(compression_header.blocksize);
+
+		compressed_data_buffer = (unsigned char *)kmm_malloc(compression_header.blocksize + 5);
```

Comment:
Why to allocate 5 extra bytes?

Reply:
This is specific to lzma sdk. It uses 5 bytes for storing compression related metadata.
Accordingly, this memory allocation specific to lzma is put under a config.